### PR TITLE
Migrate examples/react/simple from CRA to Vite

### PR DESCRIPTION
**Pull request by zorzysty**

### CODE REVIEW

File: examples/react/simple/package.json [hunk 1]
```diff
@@ -3,26 +3,21 @@
   "private": true,
   "version": "0.0.1",
   "scripts": {
-    "start": "rescripts start",
-    "build": "rescripts build",
-    "test": "rescripts test",
-    "eject": "rescripts eject"
+    "dev": "vite",
+    "start": "vite",
```
Comment (zorzysty) [inline examples/react/simple/package.json:7]: `dev` script is the default way of running Vite apps. `npm run dev` is also the command used in the root of tanstack query to run dev env for packages. 

So I wonder whether to remove the `start` script or just leave it here for "historical" reasons.

Comment (TkDodo) [reply examples/react/simple/package.json:7]: it depends what codesandbox picks up per default 🤔

Comment (zorzysty) [reply examples/react/simple/package.json:7]: I believe it works fine with `dev`: https://codesandbox.io/s/qrf93

Comment (TkDodo) [reply examples/react/simple/package.json:7]: Then let's remove it

Comment (zorzysty) [reply examples/react/simple/package.json]: [done](https://github.com/TanStack/query/pull/3912/commits/a8ae567022fbee339150784895737feb9b3f07b9 )

**Pull request merged**